x/mailbox/keeper/msg_server.go

```diff
@@ -106,21 +104,23 @@ func (k Keeper) Dispatch(goCtx context.Context, msg *types.MsgDispatch) (*types.
 	}
 	// Store that the leaf
 	store.Set(types.MailboxIMTKey(k.Tree.Count()-1), id)
-	hexSender := hexutil.Encode(sender)
 
 	// Emit the events
 	ctx.EventManager().EmitEvents(sdk.Events{
 		sdk.NewEvent(
 			types.EventTypeDispatch,
-			sdk.NewAttribute(types.AttributeKeySender, hexSender),
-			sdk.NewAttribute(types.AttributeKeyDestinationDomain, strconv.FormatUint(uint64(msg.DestinationDomain), 10)),
-			sdk.NewAttribute(types.AttributeKeyRecipientAddress, msg.RecipientAddress),
-			sdk.NewAttribute(types.AttributeKeyMessage, msg.MessageBody),
+			sdk.NewAttribute(types.AttributeKeyDestination, strconv.FormatUint(uint64(msg.DestinationDomain), 10)),
 			sdk.NewAttribute(types.AttributeKeyHyperlaneMessage, hyperlaneMsg),
+			sdk.NewAttribute(types.AttributeKeyMessage, msg.MessageBody),
+			sdk.NewAttribute(types.AttributeKeyNonce, strconv.FormatUint(uint64(nonce), 10)),
+			sdk.NewAttribute(types.AttributeKeyOrigin, strconv.FormatUint(uint64(origin), 10)),
+			sdk.NewAttribute(types.AttributeKeyRecipientAddress, msg.RecipientAddress),
+			sdk.NewAttribute(types.AttributeKeySender, msg.Sender),
+			sdk.NewAttribute(types.AttributeKeyVersion, strconv.FormatUint(0, 10)), // TODO(nix): How to determine version?
```

Comment:
Is there a penalty for adding too many attributes?

Reply:
What does the new `types.AttributeKeyHyperlaneMessage` give us?

Reply:
I refactored back to `hexSender`

Reply:
> The idea is that you can get all of the values from the hyperlaneMsg

Worth further discussion. It's a concatenated byte array AFAICT which seems hard to deserialize.